python/nav/jwtconf.py

```diff
@@ -11,32 +11,59 @@ class JWTConf(NAVConfigParser):
     """jwt.conf config parser"""
 
     DEFAULT_CONFIG_FILES = ('jwt.conf',)
+    NAV_SECTION = "nav-config"
+    DEFAULT_CONFIG = u"""
+[nav-config]
+private_key=
+public_key=
+name=
+"""
 
     def get_issuers_setting(self):
-        issuers_settings = dict()
-        for section in self.sections():
-            try:
+        try:
+            external_settings = self._get_settings_for_external_tokens()
+            local_settings = self._get_settings_for_nav_issued_tokens()
+            external_settings.update(local_settings)
+            return external_settings
+        except ConfigurationError as error:
+            _logger.error('Error reading jwtconfig {}'.format(error))
+            return dict()
+
+    def _get_settings_for_external_tokens(self):
+        settings = dict()
+        try:
+            for section in self.sections():
+                if section == self.NAV_SECTION:
+                    continue
                 get = partial(self.get, section)
+                issuer = self._validate_issuer(section)
                 key = self._validate_key(get('key'))
                 aud = self._validate_audience(get('aud'))
                 key_type = self._validate_type(get('keytype'))
                 if key_type == 'PEM':
-                    key = self._read_file(key)
+                    key = self._read_key_from_path(key)
                 claims_options = {
                     'aud': {'values': [aud], 'essential': True},
                 }
-                issuers_settings[section] = {
+                settings[issuer] = {
                     'key': key,
                     'type': key_type,
                     'claims_options': claims_options,
                 }
-            except (configparser.Error, ConfigurationError) as error:
-                _logger.error('Error collecting stats for %s: %s', section, error)
-        return issuers_settings
+        except (
+            configparser.Error,
+            configparser.NoSectionError,
+            configparser.NoOptionError,
+        ) as error:
+            raise ConfigurationError(error)
+        return settings
 
-    def _read_file(self, file):
-        with open(file, "r") as f:
-            return f.read()
+    def _read_key_from_path(self, path):
+        try:
+            with open(path, "r") as f:
+                return f.read()
+        except FileNotFoundError:
+            raise ConfigurationError("Could not read PEM key from file {}".format(path))
```

Comment:
A just as likely scenario here is that the file exists, but we lack the privileges needed to read it (happens really fast when you're trying to hide your secret files from the wrong people).

That case should also be handled, and a test should be added to test this behavior.